src/server/registry/handler.go

```diff
@@ -39,17 +41,17 @@ func New(url *url.URL) http.Handler {
 	rootRouter.Path("/v2/_catalog").Methods(http.MethodGet).Handler(catalog.NewHandler())
 
 	// handle list tag
-	rootRouter.Path("/v2/{name}/tags/list").Methods(http.MethodGet).Handler(tag.NewHandler())
+	rootRouter.Path("/v2/{project}/{repository:.*}/tags/list").Methods(http.MethodGet).Handler(tag.NewHandler())
```

Comment:
Let's not make such change.
With this change, a push to `/ubuntu:14.04`(no project) will get a unexpected status code?

Reply:
But our new artifact APIs are all `/v2/projects/{project_id}/repositories/{repository_id}/...`

Reply:
It contains `project_id`

Reply:
Yes, but let's try not to create new debt

Reply:
Updated